libraries/AC_Fence/AC_Fence.cpp

```diff
@@ -350,7 +350,9 @@ bool AC_Fence::pre_arm_check(const char* &fail_msg) const
 
     // check no limits are currently breached
     if (_breached_fences) {
-        fail_msg =  "vehicle outside fence";
+        static char msg[25];
```

Comment:
This burns 25 bytes of RAM *all the time*.  Not really acceptable.

If you really want to do this then a buffer+length will need to be passed in - like we do for most pre-arm methods.